### PR TITLE
*: solve cloud tikv disk full problem

### PR DESCRIPTION
**Pull request by tier-cap**
### What problem does this PR solve?

Issue Number：https://github.com/tikv/tikv/issues/10537

When the disk is full, no matter one or two or the majority or all, what we expected will be:

1. TiKV not crash directly.
2. Support restoring storage through drop table/delete from table, expansion is also allowed.
3. Provide slightly better reading ability.

###  What is changed and how it works?

1. Add operations options when TiKV disk full happens.
2. Add disk full error code.

### Release note

- No release note.

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -686,9 +686,9 @@ message Context {
     // Any additional serialized information about the request.
     bytes resource_group_tag = 18;
 
-    // Flag used to tell tikv that operations can be allowed even when tikv disk full happens.
-    bool allowed_on_disk_full = 19;
-
+    // Used to tell TiKV what operations are allowed when the disk status is at different levels.
+    AllowedLevel allowed_level = 19;
```
Comment (disksing) [inline proto/kvrpcpb.proto:690]: This is too ambiguous. We can tell what is it used for from the name.

Comment (disksing) [reply proto/kvrpcpb.proto:690]: Maybe consider `DiskFullOpt`

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -766,6 +766,13 @@ enum IsolationLevel {
     RC = 1; // RC = read committed
 }
 
+//Operation allowed level during each TiKV disk full threshold.
+enum AllowedLevel {
+    Allowed_Normal = 0; // Normal is the default value, meaning all operations are allowed when disk status is normal.
```
Comment (hicqu) [inline proto/kvrpcpb.proto:771]: It's better to be `Normal`, `AllowedOnAlmostFull` and `AllowedOnAlreadyFull`.

File: proto/kvrpcpb.proto [hunk 3]
```diff
@@ -766,6 +766,13 @@ enum IsolationLevel {
     RC = 1; // RC = read committed
 }
 
+//Operation allowed level during each TiKV disk full threshold.
```
Comment (hicqu) [inline proto/kvrpcpb.proto:769]: Generally there should be a space after `//`.

**Comment by disksing** [comment]
@tier-cap  want to update the naming?

File: proto/kvrpcpb.proto (hunk 2 above)
Comment (tier-cap) [reply proto/kvrpcpb.proto:771]: changed

File: proto/kvrpcpb.proto (hunk 1 above)
Comment (tier-cap) [reply proto/kvrpcpb.proto:690]: changed

File: proto/kvrpcpb.proto [hunk 4]
```diff
@@ -766,6 +766,13 @@ enum IsolationLevel {
     RC = 1; // RC = read committed
 }
 
+// Operation allowed info during each TiKV storage threshold.
+enum DiskFullOpt {
+    AllowedOnNormal = 0; // the default value, means all operations are allowed when disk status is normal.
```
Comment (5kbpers) [inline proto/kvrpcpb.proto:771]:
```suggestion
    Normal = 0; // the default value, means all operations are allowed when disk status is normal.
```

**Review by 5kbpers — COMMENTED** [review]
rest LGTM

File: proto/kvrpcpb.proto (hunk 4 above)
Comment (tier-cap) [reply proto/kvrpcpb.proto:771]: CommandPri::Normal

Comment (hicqu) [reply proto/kvrpcpb.proto:771]: I prefer `Normal` too. I think the compile error is not related to this change.

Comment (hicqu) [reply proto/kvrpcpb.proto:771]: I know why `Normal` is bad. So please change to `Default`.

**Review by 5kbpers — APPROVED** [review]

File: proto/kvrpcpb.proto (hunk 4 above)
Comment (tier-cap) [reply proto/kvrpcpb.proto]: ok, done!

**Review by disksing — APPROVED** [review]

**Comment by disksing** [comment]
@tier-cap Please resolver conflicts so I can merge it.

File: proto/kvrpcpb.proto (hunk 3 above)
Comment (tier-cap) [reply proto/kvrpcpb.proto]: done!

**Review by hicqu — APPROVED** [review]

**Pull request merged**